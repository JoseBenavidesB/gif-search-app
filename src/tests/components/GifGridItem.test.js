import React from 'react'
import { GifGridItem } from "../../components/GifGridItem";


import {shallow} from 'enzyme';

describe('Test <GifGridItem/>', () => { 
    
    const title = 'This is a test';
    const url = 'https://localhost.com'
    const wrapper = shallow( <GifGridItem 
        title={title} 
        url= {url}/> );
        
    test('should <GifGridItem/> correctly', () => {  

        expect( wrapper ).toMatchSnapshot();
    });

    test('should have paragraph with title', () => {  

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title);
    });

    test('should have image equal url and alt equal title', ()=> {

        const img = wrapper.find('img')

        expect( img.prop('src') ).toBe(url);
        expect( img.props().alt ).toBe( title );

    });

    test('should have animate__headShake class', ()=> {

        const className = 'animate__headShake';
        const div = wrapper.find('div');

        expect(div.prop('className').includes(className)).toBe(true);
    });

});