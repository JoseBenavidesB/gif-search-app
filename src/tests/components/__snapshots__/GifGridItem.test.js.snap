// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem/> should <GifGridItem/> correctly 1`] = `
<div
  className="card animate__animated animate__headShake"
>
  <img
    alt="This is a test"
    src="https://localhost.com"
  />
  <p>
     
    This is a test
     
  </p>
</div>
`;
